python/hsfs/feature_group.py

```diff
@@ -2049,6 +2059,24 @@ def online_enabled(self) -> bool:
     def online_enabled(self, online_enabled: bool) -> None:
         self._online_enabled = online_enabled
 
+    @property
+    def path(self) -> Optional[str]:
+        return self._path
+
+    @property
+    def storage_connector(self) -> "sc.StorageConnector":
+        return self._storage_connector
+
+    def get_uri(self) -> str:
+        """Location of data."""
+        if (self.storage_connector is None):
+            return self.location
+        else:
+            path = self.storage_connector._get_path(self.path)
+            if engine.get_type().startswith("spark"):
+                path = engine.get_instance().setup_storage_connector(self.storage_connector, path)
+            return path
```

Comment:
```suggestion
            return self.storage_connector._get_path(path)
```

Reply:
we have to prepare spark for the storage connector.

Reply:
it should be done when you actually read data, right?

Reply:
yeah, but `prepare_spark` also updates the URI. It changes `s3://` to `s3a://` so i hoped to avoid doing additional checks everywhere, since it is needed not only for `read`, but also for operations `write`, `register_temporary_table`, `append_column`, and `delete_record`.

Reply:
ok, but then the method name can be more explicit. Also, we should return the path for python client as well.

Reply:
i updated name and now i use location received from the back-end. This means more testing will be needed and ill check if there are no issues to just return `s3a://` from hopsworks instead of `s3://` and then changing it in the api.